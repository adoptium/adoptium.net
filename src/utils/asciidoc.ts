import Asciidoctor from "@asciidoctor/core";
import fs from "fs";
import path from "path";
import { decode } from "html-entities";

interface AsciidoctorDocument {
  getSourceLocation(): { getPath(): string };
}

interface AsciidoctorReader {
  pushInclude(
    content: string,
    file: string,
    target: string,
    line: number,
    attributes: Record<string, unknown>
  ): void;
}

interface AsciidoctorAttributes {
  [key: string]: unknown;
}

// Initialize Asciidoctor processor
const asciidoctor = Asciidoctor();

/**
 * Custom Asciidoctor include processor that resolves partials
 */
class CustomIncludeProcessor {
  private basePath: string;

  constructor(basePath: string) {
    this.basePath = basePath;
  }

  process(
    doc: AsciidoctorDocument,
    reader: AsciidoctorReader,
    target: string,
    attributes: AsciidoctorAttributes
  ) {
    // Get current file directory to resolve relative includes correctly
    const currentDir = path.dirname(doc.getSourceLocation().getPath());

    let includeFile = target;

    // If target is a relative path, resolve it from current file directory
    if (!path.isAbsolute(target) && !target.startsWith("/")) {
      includeFile = path.resolve(currentDir, target);
    }

    // If not found, try from the content base path
    if (!fs.existsSync(includeFile)) {
      includeFile = path.join(this.basePath, target);
    }

    if (fs.existsSync(includeFile)) {
      const content = fs.readFileSync(includeFile, "utf8");
      return reader.pushInclude(content, includeFile, target, 1, attributes);
    } else {
      console.warn(`Include file not found: ${includeFile}`);
      return reader.pushInclude(
        `[WARNING] Include file not found: ${target}`,
        target,
        target,
        1,
        attributes
      );
    }
  }
}

/**
 * Process AsciiDoc content with proper include resolution
 */
export function processAsciiDoc(
  filePath: string,
  content: string,
  options: Record<string, unknown> = {}
) {
  const basePath = path.dirname(filePath);
  const registry = asciidoctor.Extensions.create();

  // Create an instance of our custom processor
  const customProcessor = new CustomIncludeProcessor(basePath);

  // Try using a simple approach first - just skip extension registration
  // and see if the document still processes correctly
  const skipExtensions = true;

  if (!skipExtensions) {
    // Register include processor using a factory function
    const IncludeProcessor = function () {};
    IncludeProcessor.prototype.handles = function () {
      return true;
    };
    IncludeProcessor.prototype.process = function (
      doc: AsciidoctorDocument,
      reader: AsciidoctorReader,
      target: string,
      attrs: AsciidoctorAttributes
    ) {
      return customProcessor.process(doc, reader, target, attrs);
    };

    registry.includeProcessor(IncludeProcessor);
  }

  // Set default options with proper typing
  const defaultOptions: Record<string, unknown> = {
    safe: "server",
    base_dir: basePath,
    attributes: {
      "source-highlighter": "highlight.js",
      icons: "font",
      sectanchors: "",
      idprefix: "",
      idseparator: "-",
    },
  };

  // Add extension registry only if we're not skipping extensions
  if (!skipExtensions) {
    defaultOptions.extension_registry = registry;
  }

  // Merge options
  const mergedOptions = { ...defaultOptions, ...options };

  // Convert AsciiDoc to HTML - use the processor without extension registration
  // if we're having issues with extensions
  try {
    let html = asciidoctor.convert(content, mergedOptions) as string;
    // Post-process image src paths: replace src="filename.png" with src="/docs/filename.png" if not already absolute or external
    html = html.replace(
      /<img([^>]+)src=["'](?![a-z]+:|\/)([^"'>]+)["']/gi,
      '<img$1src="/docs/$2"'
    );
    return html;
  } catch (error) {
    console.error("Error converting AsciiDoc content:", error);

    // Try without extension registry as fallback
    // Create a new options object without the extension_registry property using destructuring
    // Use object destructuring with "rest" syntax to remove the property
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { extension_registry, ...fallbackOptions } = defaultOptions;
    let html = asciidoctor.convert(content, fallbackOptions) as string;
    html = html.replace(
      /<img([^>]+)src=["'](?![a-z]+:|\/)([^"'>]+)["']/gi,
      '<img$1src="/docs/$2"'
    );
    return html;
  }
}

/**
 * Decode HTML entities in a string
 */
export function decodeHtmlEntities(text: string): string {
  return decode(text);
}

/**
 * Extract metadata from AsciiDoc document
 */
export function extractMetadata(
  filePath: string,
  content: string
): {
  title: string;
  description: string;
  authors: string[];
  attributes: Record<string, unknown>;
} {
  const doc = asciidoctor.load(content, { safe: "server" });

  // Get the title and ensure it's a string
  const docTitle = doc.getDocumentTitle();
  let title =
    typeof docTitle === "string"
      ? docTitle
      : docTitle?.toString() || path.basename(filePath, path.extname(filePath));

  // Decode HTML entities in the title
  title = decodeHtmlEntities(title);

  // Get and decode description if it exists
  const description = doc.hasAttribute("description")
    ? decodeHtmlEntities(String(doc.getAttribute("description")))
    : "";

  return {
    title,
    description,
    authors: doc.hasAttribute("page-authors")
      ? String(doc.getAttribute("page-authors"))
          .split(",")
          .map((author: string) => author.trim())
      : [],
    attributes: doc.getAttributes() as Record<string, unknown>,
  };
}

/**
 * Check if a file exists
 */
export function fileExists(filePath: string): boolean {
  try {
    return fs.existsSync(filePath);
  } catch {
    // Silently handle any errors and return false
    return false;
  }
}

/**
 * Get all available language variants for a given base path
 */
export function getLanguageVariants(
  basePath: string,
  fileName: string
): string[] {
  const dir = path.dirname(basePath);
  const baseFileName = path.basename(fileName, ".adoc");

  // Get all adoc files in the directory
  const files = fs
    .readdirSync(dir)
    .filter((file) => file.startsWith(baseFileName) && file.endsWith(".adoc"));

  // Extract locales from filenames
  return files.map((file) => {
    // Format: index.de.adoc -> de
    const match = file.match(
      new RegExp(`${baseFileName}(?:\\.([\\w-]+))?\\.adoc$`)
    );
    return match && match[1] ? match[1] : "en";
  });
}
