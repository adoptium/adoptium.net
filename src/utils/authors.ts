import AuthorData from "@/data/authors.json";

export interface Author {
  name: string;
  summary?: string;
  twitter?: string;
  github?: string;
  linkedin?: string;
}

export function getFormattedAuthorData(authorId: string): Author {
  const rawAuthorData = AuthorData[authorId as keyof typeof AuthorData];

  if (!rawAuthorData) {
    return {
      name: "Unknown Author",
    };
  }

  // Convert the raw author data to our Author interface
  return {
    name: rawAuthorData.name || "Unknown Author",
    // Convert null values to undefined
    summary: rawAuthorData.summary || undefined,
    twitter: rawAuthorData.twitter || undefined,
    github: rawAuthorData.github || undefined,
    linkedin: rawAuthorData.linkedin || undefined,
  };
}
