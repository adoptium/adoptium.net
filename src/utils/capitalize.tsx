// Format text with a capital letter
export function capitalize(text: string) {
  switch (text) {
    case "alpine-linux":
      return "Alpine Linux"
    case "alpine_linux":
      return "Alpine Linux"
    case "redhat":
      return "Red Hat"
    case "ibm":
      return "IBM"
    case "mac":
      return "macOS"
    case "semeru":
      return "Semeru Runtimes"
    default:
      return text.charAt(0).toUpperCase() + text.slice(1)
  }
}
