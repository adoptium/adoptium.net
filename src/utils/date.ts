export function formatDate(
  date: string = "",
  locale: string = "en-US"
): string {
  const targetDate = date ? new Date(date) : new Date();
  return targetDate.toLocaleDateString(locale, {
    month: "long",
    day: "2-digit",
    year: "numeric",
  });
}
