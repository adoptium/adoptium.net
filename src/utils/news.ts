import { getBlogPosts } from "./markdown";
import { fetchLatestNews } from "./eclipse";

interface NewsResult {
  posts: ReturnType<typeof getBlogPosts>;
  totalPages: number;
}

// Interface for Eclipse Foundation news items
interface EclipseNewsItem {
  title: string;
  date: string;
  link: string;
  body: string;
  image?: string;
}

export async function getNews({
  numPosts = 9,
  page = 1,
  includeEF = false,
} = {}): Promise<NewsResult> {
  const blogs = getBlogPosts();
  if (includeEF) {
    const efNews = await fetchLatestNews();
    // structure efNews to match the blog posts format
    const efPosts = efNews.news
      .filter(
        (newsItem: EclipseNewsItem) =>
          !newsItem.link.startsWith("https://adoptium.net")
      )
      .map((newsItem: EclipseNewsItem) => ({
        slug: newsItem.link,
        newsItem,
        metadata: {
          title: newsItem.title,
          date: newsItem.date,
          tags: ["eclipse-news"],
          author: "Eclipse Foundation",
          description: newsItem.body.substring(0, 150) + "...",
          featuredImage:
            newsItem.image && newsItem.image.trim() !== ""
              ? newsItem.image
              : "/images/backgrounds/ef-default-news.png",
        },
      }));
    // Combine the blog posts with the EF news posts
    blogs.push(...efPosts);
  }

  const sortedBlogs = blogs.sort((a, b) => {
    const dateA = new Date(a.metadata.date).getTime();
    const dateB = new Date(b.metadata.date).getTime();
    return dateB - dateA; // Sort in descending order
  });

  const totalPages = Math.ceil(sortedBlogs.length / numPosts);
  const paginatedPosts = sortedBlogs.slice(
    (page - 1) * numPosts,
    page * numPosts
  );

  return {
    posts: paginatedPosts,
    totalPages,
  };
}

export function getNewsByTag(
  tag: string,
  { numPosts = 6, page = 1 } = {}
): NewsResult {
  const blogs = getBlogPosts();
  const filteredBlogs = blogs.filter((post) =>
    post.metadata.tags?.includes(tag)
  );

  const sortedBlogs = filteredBlogs.sort((a, b) => {
    const dateA = new Date(a.metadata.date).getTime();
    const dateB = new Date(b.metadata.date).getTime();
    return dateB - dateA; // Sort in descending order
  });

  const totalPages = Math.ceil(sortedBlogs.length / numPosts);
  const paginatedPosts = sortedBlogs.slice(
    (page - 1) * numPosts,
    page * numPosts
  );

  return {
    posts: paginatedPosts,
    totalPages,
  };
}

export function getNewsByAuthor(
  author: string,
  { numPosts = 6, page = 1 } = {}
): NewsResult {
  const blogs = getBlogPosts();
  const filteredBlogs = blogs.filter((post) => post.metadata.author === author);

  const sortedBlogs = filteredBlogs.sort((a, b) => {
    const dateA = new Date(a.metadata.date).getTime();
    const dateB = new Date(b.metadata.date).getTime();
    return dateB - dateA; // Sort in descending order
  });

  const totalPages = Math.ceil(sortedBlogs.length / numPosts);
  const paginatedPosts = sortedBlogs.slice(
    (page - 1) * numPosts,
    page * numPosts
  );

  return {
    posts: paginatedPosts,
    totalPages,
  };
}
