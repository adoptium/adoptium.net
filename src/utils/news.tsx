import { getBlogPosts } from "./markdown";

interface NewsResult {
  posts: ReturnType<typeof getBlogPosts>;
  totalPages: number;
}

export function getNews({numPosts = 6, page = 1} = {}): NewsResult {
  const blogs = getBlogPosts();
  const sortedBlogs = blogs.sort((a, b) => {
    const dateA = new Date(a.metadata.date).getTime();
    const dateB = new Date(b.metadata.date).getTime();
    return dateB - dateA; // Sort in descending order
  });
  
  const totalPages = Math.ceil(sortedBlogs.length / numPosts);
  const paginatedPosts = sortedBlogs.slice((page - 1) * numPosts, page * numPosts);
  
  return {
    posts: paginatedPosts,
    totalPages
  };
}

export function getNewsByTag(tag: string, {numPosts = 6, page = 1} = {}): NewsResult {
  const blogs = getBlogPosts();
  const filteredBlogs = blogs.filter(post => post.metadata.tags?.includes(tag));
  
  const sortedBlogs = filteredBlogs.sort((a, b) => {
    const dateA = new Date(a.metadata.date).getTime();
    const dateB = new Date(b.metadata.date).getTime();
    return dateB - dateA; // Sort in descending order
  });
  
  const totalPages = Math.ceil(sortedBlogs.length / numPosts);
  const paginatedPosts = sortedBlogs.slice((page - 1) * numPosts, page * numPosts);
  
  return {
    posts: paginatedPosts,
    totalPages
  };
}

export function getNewsByAuthor(author: string, { numPosts = 6, page = 1 } = {}): NewsResult {
  const blogs = getBlogPosts();
  const filteredBlogs = blogs.filter(post => post.metadata.author === author);

  const sortedBlogs = filteredBlogs.sort((a, b) => {
    const dateA = new Date(a.metadata.date).getTime();
    const dateB = new Date(b.metadata.date).getTime();
    return dateB - dateA; // Sort in descending order
  });

  const totalPages = Math.ceil(sortedBlogs.length / numPosts);
  const paginatedPosts = sortedBlogs.slice((page - 1) * numPosts, page * numPosts);

  return {
    posts: paginatedPosts,
    totalPages
  };
}
