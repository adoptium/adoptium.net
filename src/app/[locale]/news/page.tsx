import React from 'react';
import { Metadata } from 'next';
import { getNews } from '@/utils/news'
import PageHeader from '@/components/Common/PageHeader';
import NewsCardList from '@/components/News/NewsCardList';

export const metadata: Metadata = {
  title: "News & Updates",
  description: "Latest news and updates from the Eclipse Adoptium Project",
}


export default function NewsPage() {
    const { posts, totalPages } = getNews({ numPosts: 6, page: 1 });
    if (!posts || posts.length === 0) {
        return <div>No news articles available.</div>;
    }
    return (
        <div>
            <PageHeader
                subtitle="News"
                title="News & Updates"
                description="Follow the latest updates from the Eclipse Adoptium Project"
                className="mx-auto max-w-[860px] px-2 w-full"
            />
            <NewsCardList
                posts={posts}
                previousPageNumber={null}
                previousPageLink={null}
                nextPage={totalPages > 1 ? '/news/page/2' : ''}
                currentPage={1}
                totalPages={totalPages}
                baseUrl={'/news'}
            />
        </div>
    )
}
