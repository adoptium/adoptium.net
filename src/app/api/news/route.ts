import { NextRequest } from "next/server";
import { getNews } from "@/utils/news";

export async function GET(req: NextRequest) {
  try {
    const { searchParams } = new URL(req.url);
    const numPosts = parseInt(searchParams.get("numPosts") || "4", 10);
    const page = parseInt(searchParams.get("page") || "1", 10);
    const includeEF = searchParams.get("includeEF") === "true";
    const news = await getNews({ numPosts, page, includeEF });
    return Response.json(news, {
      headers: {
        "Cache-Control": "public, max-age=3600, stale-while-revalidate=60",
      },
    });
  } catch {
    return Response.json({ error: "Failed to fetch news" }, { status: 500 });
  }
}
