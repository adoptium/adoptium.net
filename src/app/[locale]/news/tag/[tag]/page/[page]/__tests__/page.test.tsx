import { describe, it, expect, vi, afterEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import { axe } from 'vitest-axe';
import TaggedNewsPage, { generateMetadata } from '../page';
import * as newsUtils from '@/utils/news';
import { notFound, redirect } from 'next/navigation';

vi.mock('@/utils/news');
vi.mock('next/navigation', () => ({ notFound: vi.fn(), redirect: vi.fn() }));

const mockGetNewsByTag = newsUtils.getNewsByTag as unknown as ReturnType<typeof vi.fn>;

describe('TaggedNewsPage', () => {
    afterEach(() => {
        vi.resetAllMocks();
    });

    it('redirects to tag root if page is 1', async () => {
        mockGetNewsByTag.mockReturnValue({ posts: [], totalPages: 1 });
        await TaggedNewsPage({ params: Promise.resolve({ tag: 'temurin', page: '1' }) });
        expect(redirect).toHaveBeenCalledWith('/news/tag/temurin');
    });

    it('renders news and header for tag and page when posts exist', async () => {
        mockGetNewsByTag.mockReturnValue({
            posts: [
                {
                    metadata: {
                        title: 'Tagged News',
                        description: 'desc',
                        date: '2025-06-18',
                        author: 'author',
                        tags: ['temurin']
                    },
                    slug: 'tagged-news'
                }
            ],
            totalPages: 3
        });
        const element = await TaggedNewsPage({ params: Promise.resolve({ tag: 'temurin', page: '2' }) });
        const { container } = render(element);
        expect(screen.getByText('News tagged with "temurin"')).toBeInTheDocument();
        expect(screen.getByText('Tagged News')).toBeInTheDocument();
        // Snapshot test
        expect(container).toMatchSnapshot();
        // Axe accessibility lint
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });

    it('calls notFound if no posts for tag and page', async () => {
        mockGetNewsByTag.mockReturnValue({ posts: [], totalPages: 1 });
        await TaggedNewsPage({ params: Promise.resolve({ tag: 'temurin', page: '2' }) });
        expect(notFound).toHaveBeenCalled();
    });

    it('generateMetadata returns correct metadata', async () => {
        const meta = await generateMetadata({ params: Promise.resolve({ tag: 'temurin', page: '2' }) });
        expect(meta).toEqual({
            title: 'News with tag: temurin - Page 2',
            description: 'News and updates from Eclipse Adoptium Project tagged with temurin',
        });
    });
});
