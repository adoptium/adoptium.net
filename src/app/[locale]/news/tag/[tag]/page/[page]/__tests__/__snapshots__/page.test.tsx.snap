// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaggedNewsPage > renders news and header for tag and page when posts exist 1`] = `
<div>
  <div>
    <div
      class="relative pt-20 pb-8 md:pb-16 md:pt-32 overflow-hidden"
    >
      <div
        class="w-full relative"
      >
        <div
          class="absolute z-[-1] w-[2396px] h-[1340px] left-[23px] md:left-[112px] top-[21px] md:top-[71px] bg-[#410170] shadow-[0_0_400px_rgba(65,1,112,1)] rounded-full blur-[400px]"
        />
        <div
          class="absolute z-[-1] w-[1487px] h-[893px] left-[120px] md:left-[676px] top-[60px] md:top-[195px] bg-[#B62987] shadow-[0_0_400px_rgba(182,41,135,1)] rounded-full blur-[400px]"
        />
        <div
          class="absolute z-[-1] w-[688px] h-[446px] left-[400px] md:left-[1131px] top-[318px] bg-[#FE8492] shadow-[0_0_400px_rgba(254,132,146,1)] rounded-full blur-[400px]"
        />
      </div>
      <div
        class="mx-auto max-w-[1048px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
      >
        <div
          class="self-stretch flex-col justify-center items-center gap-6 flex"
        >
          <div
            class="self-stretch flex-col justify-center items-center gap-4 flex"
          >
            <div
              class="justify-start items-center gap-3 inline-flex"
            >
              <svg
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#FF1464"
                  height="8"
                  rx="2"
                  width="8"
                  x="3"
                  y="3"
                />
                <rect
                  height="11"
                  rx="3.5"
                  stroke="#FF1464"
                  stroke-opacity="0.25"
                  stroke-width="3"
                  width="11"
                  x="1.5"
                  y="1.5"
                />
              </svg>
              <div
                class="text-rose-600 text-base font-bold leading-normal"
              >
                News
              </div>
            </div>
            <div
              class="self-stretch text-center text-white text-[56px] lg:text-[80px] leading-[114.286%] md:leading-[120%] font-semibold"
            >
              News tagged with "temurin"
            </div>
          </div>
          <div
            class="self-stretch text-center text-lightgrey text-[20px] font-normal leading-[140%] mx-auto max-w-[860px] px-2 w-full"
          >
            Follow the latest updates from the Eclipse Adoptium Project
          </div>
        </div>
      </div>
    </div>
    <div
      class="max-w-[1264px] mx-auto px-6 py-8 md:pt-12"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 justify-items-center py-8 md:py-12"
      >
        <div
          class="flex flex-col max-w-[385px] w-full sm:w-[385px] min-h-[600px] rounded-3xl newscard-2 bg-[#200D46]"
        >
          <img
            alt="blog banner image"
            class="m-0.5 rounded-t-3xl h-[200px] object-cover w-full"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fnews%2Fundefined%2Fundefined%2Ftagged-news%2Fopengraph-image&w=828&q=75"
            srcset="/_next/image?url=%2Fnews%2Fundefined%2Fundefined%2Ftagged-news%2Fopengraph-image&w=640&q=75 1x, /_next/image?url=%2Fnews%2Fundefined%2Fundefined%2Ftagged-news%2Fopengraph-image&w=828&q=75 2x"
            style="color: transparent;"
            width="385"
          />
          <div
            class="p-8 flex flex-col h-[400px]"
          >
            <h3
              class="text-[24px] font-semibold leading-normal h-auto line-clamp-3 mb-2"
              title="Tagged News"
            >
              Tagged News
            </h3>
            <div
              class="flex justify-between py-2"
            >
              <h3
                class="flex flex-col gap-1 tab-button-text text-white"
              >
                <span>
                  June 18, 2025
                </span>
              </h3>
            </div>
            <div
              class="flex-grow overflow-hidden mb-4"
            >
              <p
                class="line-clamp-4 text-base text-gray-200"
              >
                desc
              </p>
            </div>
            <div
              class="mt-auto"
            >
              <a
                href="[object Object]"
              >
                <button
                  class="rounded-2xl bg-transparent gradient-border border-2 border-pink-500/0 text-white text-base leading-6 font-bold w-[160px] h-[48px] block"
                >
                  Read More
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-center items-center gap-8 md:gap-14 my-8"
      >
        <a
          class="flex items-center gap-3"
          href="/news/tag/temurin"
          rel="prev"
        >
          <span
            class="cursor-pointer"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
              />
            </svg>
          </span>
          <p
            class="tab-button-text mb-0 hidden md:block"
          >
            Previous
          </p>
        </a>
        <div
          class="flex items-center gap-5"
        >
          <div
            class="flex items-center justify-between gap-6 border border-[#3E3355] rounded-[80px] px-6 py-3"
          >
            <a
              href="/news/tag/temurin"
            >
              <p
                class="tab-button-text mb-0 cursor-pointer "
              >
                1
              </p>
            </a>
            <a
              href="/news/tag/temurin/page/2"
            >
              <p
                class="tab-button-text mb-0 cursor-pointer font-bold text-pink"
              >
                2
              </p>
            </a>
            <a
              href="/news/tag/temurin/page/3"
            >
              <p
                class="tab-button-text mb-0 cursor-pointer "
              >
                3
              </p>
            </a>
          </div>
        </div>
        <a
          class="flex items-center gap-3"
          href="/news/tag/temurin/page/3"
        >
          <p
            class="tab-button-text mb-0 hidden md:block"
          >
            Next
          </p>
          <span
            class="cursor-pointer"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
              />
            </svg>
          </span>
        </a>
      </div>
      <div
        class="flex justify-center items-center gap-5 mt-8"
      >
        <a
          class="bg-primary hover:bg-primary-dark text-white py-3 px-6 rounded-full transition-all duration-200 ease-in-out text-center font-medium"
          href="https://newsroom.eclipse.org/node/add/news"
          rel="noopener noreferrer"
          target="_blank"
        >
          Submit News
        </a>
      </div>
    </div>
  </div>
</div>
`;
